code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm

```diff
@@ -263,71 +282,66 @@ Difficulty: Medium
 	density = TRUE
 	sleep(1)
 	swooping &= ~SWOOP_DAMAGEABLE
-	SetRecoveryTime(MEGAFAUNA_DEFAULT_RECOVERY_TIME)
+	swoop_cooldown = world.time + cooldown
+	
+/mob/living/simple_animal/hostile/megafauna/dragon/proc/goto_target()
+	if(!target)
+		return
+	forceMove(get_turf(target))
 
 /mob/living/simple_animal/hostile/megafauna/dragon/AltClickOn(atom/movable/A)
 	if(!istype(A))
 		return
-	if(swoop_cooldown >= world.time)
+	if(player_cooldown >= world.time)
 		to_chat(src, "<span class='warning'>You need to wait 20 seconds between swoop attacks!</span>")
 		return
-	swoop_attack(TRUE, A, 25)
+	swoop_attack(TRUE, A, 30)
+	player_cooldown = world.time + 200
 
 /obj/item/gps/internal/dragon
 	icon_state = null
 	gpstag = "Fiery Signal"
 	desc = "Here there be dragons."
 	invisibility = 100
 
-
-/obj/effect/temp_visual/fireball
-	icon = 'icons/obj/wizard.dmi'
-	icon_state = "fireball"
-	name = "fireball"
-	desc = "Get out of the way!"
-	layer = FLY_LAYER
-	randomdir = FALSE
-	duration = 9
-	pixel_z = DRAKE_SWOOP_HEIGHT
-
-/obj/effect/temp_visual/fireball/Initialize()
-	. = ..()
-	animate(src, pixel_z = 0, time = duration)
-
 /obj/effect/temp_visual/target
```

Comment:
you should rename this to lava_warning or something like that, target is a bit too generic

Reply:
Changing this to that actually did bring up the issue of the admin punishment using the drake effect for a fireball, should I move the old drake fireball code to the admin punishment file or just keep it in the drake file?